src/jit/morph.cpp

```diff
@@ -17958,45 +17970,58 @@ void Compiler::fgMarkDemotedImplicitByRefArgs()
     {
         LclVarDsc* varDsc = &lvaTable[lclNum];
 
-        if (lvaIsImplicitByRefLocal(lclNum) && varDsc->lvPromoted)
-        {
-            // We stashed the pointer to the real promotion temp in lvFieldLclStart
-            unsigned structLclNum = varDsc->lvFieldLclStart;
-
-            // Unmark the parameter as promoted.
-            varDsc->lvPromoted      = false;
-            varDsc->lvFieldCnt      = 0;
-            varDsc->lvFieldLclStart = 0;
-            // Clear its ref count; this was set during address-taken analysis so that
-            // call morphing could identify single-use implicit byrefs; we're done with
-            // that, and want it to be in its default state of zero when we go to set
-            // real ref counts for all variables.
-            varDsc->lvRefCnt = 0;
-
-            // The temp struct is now unused; set flags appropriately so that we
-            // won't allocate space for it on the stack.
-            LclVarDsc* structVarDsc     = &lvaTable[structLclNum];
-            structVarDsc->lvRefCnt      = 0;
-            structVarDsc->lvAddrExposed = false;
+        if (lvaIsImplicitByRefLocal(lclNum))
+        {
+            if (varDsc->lvPromoted)
+            {
+                // The parameter is simply a pointer now, so clear lvPromoted.
```

Comment:
This is something that could be confusing if viewed in isolation. You might also mention here that this field has been used temporarily to communicate between `fgRetypeImplicitByRefArgs()` and this method.